src/vs/workbench/contrib/debug/common/loadedScriptsPicker.ts

```diff
@@ -0,0 +1,105 @@
+import { getIconClasses } from 'vs/editor/common/services/getIconClasses';
+import { IModelService } from 'vs/editor/common/services/model';
+import { ILanguageService } from 'vs/editor/common/languages/language';
+
+export interface IPickerLoadedScriptItem extends IQuickPickItem {
+	accept(): void;
+}
+
+
+/**
+ * This function takes a regular quickpick and makes one for loaded scripts that has persistent headers
+ * e.g. when some picks are filtered out, the ones that are visible still have its header.
+ */
+export async function showLoadedScriptMenu(sessions: IDebugSession[], quickInputService: IQuickInputService, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService) {
+
+	const quickPick = quickInputService.createQuickPick<IPickerLoadedScriptItem>();
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	const changeListener = quickPick.onDidChangeValue(async () => {
+		quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	});
+	const acceptListener = quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		acceptListener.dispose();
+		changeListener.dispose();
+		quickPick.hide();
+		quickPick.dispose();
+	});
+	quickPick.show();
+}
+
+function _getLabelName(source: Source) {
+	const posOfForwardSlash = source.name.lastIndexOf('/');
+	const posOfBackSlash = source.name.lastIndexOf('\\');
+
+	const delimeterPos = posOfForwardSlash > posOfBackSlash ? posOfForwardSlash : posOfBackSlash;
+	const label = source.name.substring(delimeterPos + 1);
+	return label;
+}
+
+async function _getPicksFromSession(session: IDebugSession, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const items: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+	items.push({ type: 'separator', label: session.name });
+	const sources = await session.getLoadedSources();
+
+	sources.forEach((element: Source) => {
+		const pick = _createPick(element, filter, editorService, modelService, languageService);
+		if (pick) {
+			items.push(pick);
+		}
+
+	});
+	return items;
+}
+async function _getPicks(filter: string, sessions: IDebugSession[], editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const loadedScriptPicks: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+
+
+	const picks = await Promise.all(
+		sessions.map((session) => {
```

Comment:
You can do this a little shorter - the parens around `session` are not required when there is only one argument, and instead of the `{}` and `return` you can just say `session => _getPicksFromSession(...)`